src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -182,14 +170,10 @@ class AdjustClipBorder final : public WaveClipAdjustBorderHandle::AdjustPolicy
                (currentTrack->GetLinkType() == Track::LinkType::Aligned && *trackList->Find(track) == currentTrack);
             for(const auto& interval : track->Intervals())
             {
-               if(isSameTrack)
+               if(isSameTrack && dynamic_cast<const WaveTrack::Interval*>(interval.get()) == &adjustedInterval)
```

Comment:
You can eliminate this ugly `dynamic_cast`, if above in the outer loop, you use `.Any<const WaveTrack>()`

Reply:
That will change the behavior as we are searching for all interval boundaries, not just in wave tracks. But it could be simplified in other way - check for track and interval equality shortens to simple track pointer comparison, as we now always iterate over leader tracks and misalligned clips is not a valid case any more